icon/Icon.ts

```diff
@@ -72,225 +72,257 @@ export const Icon = {
+    toast(p?: IconProps)                {return Icon.icon({...p, iconName: 'bread-slice'})},
+    toolbox(p?: IconProps)              {return Icon.icon({...p, iconName: 'toolbox'})},
+    tools(p?: IconProps)                {return Icon.icon({...p, iconName: 'tools'})},
+    trash(p?: IconProps)                {return Icon.icon({...p, iconName: 'trash-alt'})},
+    treeList(p?: IconProps)             {return Icon.icon({...p, iconName: 'list-tree'})},
+    undo(p?: IconProps)                 {return Icon.icon({...p, iconName: 'undo'})},
+    unlink(p?: IconProps)               {return Icon.icon({...p, iconName: 'unlink'})},
+    unlock(p?: IconProps)               {return Icon.icon({...p, iconName: 'lock-open'})},
+    user(p?: IconProps)                 {return Icon.icon({...p, iconName: 'user'})},
+    userCheck(p?: IconProps)            {return Icon.icon({...p, iconName: 'user-check'})},
+    userCircle(p?: IconProps)           {return Icon.icon({...p, iconName: 'user-circle'})},
+    userClock(p?: IconProps)            {return Icon.icon({...p, iconName: 'user-clock'})},
+    users(p?: IconProps)                {return Icon.icon({...p, iconName: 'users'})},
+    warning(p?: IconProps)              {return Icon.icon({...p, iconName: 'exclamation-triangle'})},
+    warningCircle(p?: IconProps)        {return Icon.icon({...p, iconName: 'exclamation-circle'})},
+    warningSquare(p?: IconProps)        {return Icon.icon({...p, iconName: 'exclamation-square'})},
+    window(p?: IconProps)               {return Icon.icon({...p, iconName: 'window'})},
+    wrench(p?: IconProps)               {return Icon.icon({...p, iconName: 'wrench'})},
+    x(p?: IconProps)                    {return Icon.icon({...p, iconName: 'times'})},
+    xCircle(p?: IconProps)              {return Icon.icon({...p, iconName: 'times-circle'})},
+    xHexagon(p?: IconProps)             {return Icon.icon({...p, iconName: 'times-hexagon'})},
+
+    /**
+     * Allows apps to override Hoist choices for given icons.
+     */
+    accessDenied:                   (p?: IconProps) => Icon.slashedCircle({...p}),
+    add:                            (p?: IconProps) => Icon.plus({...p}),
+    analytics:                      (p?: IconProps) => Icon.mixedChart({...p}),
+    approve:                        (p?: IconProps) => Icon.userCheck({...p}),
+    attachment:                     (p?: IconProps) => Icon.paperclip({...p}),
+    close:                          (p?: IconProps) => Icon.x({...p}),
+    columnMenu:                     (p?: IconProps) => Icon.bars({...p}),
+    danger:                         (p?: IconProps) => Icon.xCircle({...p}),
+    delete:                         (p?: IconProps) => Icon.minusCircle({...p}),
+    detail:                         (p?: IconProps) => Icon.magnifyingGlass({...p}),
+    diff:                           (p?: IconProps) => Icon.arrowRightArrowLeft({...p}),
+    disabled:                       (p?: IconProps) => Icon.slashedCircle(),
```

Comment:
I think there is an extra cloning here.  There should be no need to clone all these props here (`{...p}`).

Also, looks like the props not forwarded at all in the case of `disabled`

Reply:
I'll fix this, eager to get this in